### PR TITLE
milestone picker: tidy up the top area #1529

### LINKED ISSUE #1529: milestone picker: tidy up the top area
This area looks a bit untidy? May be remove the dotted outlines?

image

### PR DESCRIPTION
**Pull request by PSNAppz**
Fixes #1529

### CODE REVIEW

**Comment by damithc** [comment]
@PSNAppz thanks for the PR. Can you post a screenshot?

**Comment by PSNAppz** [comment]
@damithc Sorry for the late reply.

I can't see an option to open MilestonePicker.

**Comment by damithc** [comment]
Type M while an issue is selected.

-- 
Sent from a mobile device.

**Comment by PSNAppz** [comment]
milestonepicker

**Comment by damithc** [comment]
Looks a bit odd without it too. May be show the line when empty (like a place holder) but not show it if there is a milestone inside the box?

**Comment by PSNAppz** [comment]
_If there is no previous milestone it will display the dotted box_

milestonepicker

_If there is previous milestone it will  not display the dotted box_

milestonepicker2

**Comment by damithc** [comment]
@m133225 for your review

File: src/main/java/ui/components/pickers/MilestonePickerDialog.java [hunk 1]
```diff
@@ -220,7 +220,10 @@ private void updateExistingMilestone(Optional<PickerMilestone> existingMilestone
         HBox assignedMilestoneBox = createPreviouslyAssignedMilestoneBox();
         assignedMilestonePane.getChildren().add(assignedMilestoneBox);
 
-        if (!existingMilestone.isPresent()) return;
+        if (!existingMilestone.isPresent()){
+            assignedMilestoneBox.setStyle("-fx-border-radius: 3;-fx-border-style: dotted;-fx-alignment:center");
```
Comment (m133225) [inline src/main/java/ui/components/pickers/MilestonePickerDialog.java]: I think @damithc intended for the dotted lines to appear if any of the labels' spaces are empty. i.e. this should also be in `updateNewlyAssignedMilestone` to handle the case when `selectedMilestone` is not present.

File: src/main/java/ui/components/pickers/MilestonePickerDialog.java [hunk 2]
```diff
@@ -220,7 +220,10 @@ private void updateExistingMilestone(Optional<PickerMilestone> existingMilestone
         HBox assignedMilestoneBox = createPreviouslyAssignedMilestoneBox();
         assignedMilestonePane.getChildren().add(assignedMilestoneBox);
 
-        if (!existingMilestone.isPresent()) return;
+        if (!existingMilestone.isPresent()){
```
Comment (m133225) [inline src/main/java/ui/components/pickers/MilestonePickerDialog.java]: Minor issue here: There should be a space between `)` and `{`. 😛

**Comment by m133225** [comment]
Thanks for the PR! Just a recommendation as well as a minor code style change requested.

P.S. Sorry for the delay, have been researching on potential architectures that may suit our application better.

**Comment by PSNAppz** [comment]
@m133225 Thanks for the recommendations. Ready to review.

File: src/main/java/ui/components/pickers/MilestonePickerDialog.java [hunk 3]
```diff
@@ -210,8 +210,10 @@ private void updateNewlyAssignedMilestone(Optional<PickerMilestone> selectedMile
         HBox newlyAssignedMilestoneBox = createNewlyAssignedMilestoneBox();
         assignedMilestoneStatus.getChildren().add(newlyAssignedMilestoneBox);
 
-        if (!selectedMilestone.isPresent()) return;
-
+        if (!selectedMilestone.isPresent()) {
+            assignedMilestoneBox.setStyle("-fx-border-radius: 3;-fx-border-style: dotted;-fx-alignment:center");
```
Comment (m133225) [inline src/main/java/ui/components/pickers/MilestonePickerDialog.java:214]: Hmm, maybe these style strings could be refactored into `private static` `NO_BORDER_STYLE` and `DOTTED_BORDER_STYLE`?

**Comment by m133225** [comment]
Aside from the mentioned change I should have no further comments. Will go ahead and approve this first. Do make the necessary changes in the meantime. Thanks!

**Pull request closed without merging**